client/src/components/Query/Statement.jsx

```diff
@@ -74,6 +74,14 @@ class Statement extends React.Component {
     this.handleNewQuery = this.handleNewQuery.bind(this);
     this.handleCombine = this.handleCombine.bind(this);
     this.findQueryIndexForKey = this.findQueryIndexForKey.bind(this);
+    this.handleAddInstruction = this.handleAddInstruction.bind(this);
+    this.handleRemoveInstruction = this.handleRemoveInstruction.bind(this);
+    this.handleReplaceInstruction = this.handleReplaceInstruction.bind(this);
+    this.handleChangeQueryTitle = this.handleChangeQueryTitle.bind(this);
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
```

Comment:
I think this falls outside the scope of the ticket and is a good example of eager optimization.

Reply:
The lifecycle contains one line of code. This prevents useless re-render of the component and helps to improve the app performances.

You can also use `PureComponent` if you prefer.

Reply:
And this morning I explained to you already that I added this while debugging a major bug (with the drag and drop) and decided to simply leave them there. You seemed fine with this at that time.

Reply:
I mentioned that it is something we should discuss as a team and possibly eventually implement it everywhere and not just at one place.